lib/matplotlib/blocking_input.py

```diff
@@ -2,70 +2,63 @@
 
-:class:`BlockingInput`
-    creates a callable object to retrieve events in a blocking way for
-    interactive sessions
-
-:class:`BlockingKeyMouseInput`
-    creates a callable object to retrieve key or mouse clicks in a blocking
-    way for interactive sessions.
-    Note: Subclass of BlockingInput. Used by waitforbuttonpress
-
-:class:`BlockingMouseInput`
-    creates a callable object to retrieve mouse clicks in a blocking way for
-    interactive sessions.
-    Note: Subclass of BlockingInput.  Used by ginput
-
-:class:`BlockingContourLabeler`
-    creates a callable object to retrieve mouse clicks in a blocking way that
-    will then be used to place labels on a ContourSet
-    Note: Subclass of BlockingMouseInput.  Used by clabel
+`BlockingInput`
+    Creates a callable object to retrieve events in a blocking way for
+    interactive sessions.  Base class of the other classes listed here.
+
+`BlockingKeyMouseInput`
+    Creates a callable object to retrieve key or mouse clicks in a blocking
+    way for interactive sessions.  Used by `waitforbuttonpress`.
+
+`BlockingMouseInput`
+    Creates a callable object to retrieve mouse clicks in a blocking way for
+    interactive sessions.  Used by `ginput`.
+
+`BlockingContourLabeler`
+    Creates a callable object to retrieve mouse clicks in a blocking way that
+    will then be used to place labels on a `ContourSet`.  Used by `clabel`.
 """
 
-import six
+import logging
+from numbers import Integral
+
 import matplotlib.lines as mlines
+import numpy as np
 
-import logging
 
 _log = logging.getLogger(__name__)
 
 
 class BlockingInput(object):
-    """
-    Class that creates a callable object to retrieve events in a
-    blocking way.
-    """
+    """Callable for retrieving events in a blocking way."""
+
     def __init__(self, fig, eventslist=()):
         self.fig = fig
         self.eventslist = eventslist
 
     def on_event(self, event):
         """
-        Event handler that will be passed to the current figure to
-        retrieve events.
+        Event handler; will be passed to the current figure to retrieve events.
         """
-        # Add a new event to list - using a separate function is
-        # overkill for the base class, but this is consistent with
-        # subclasses
+        # Add a new event to list - using a separate function is overkill for
+        # the base class, but this is consistent with subclasses.
         self.add_event(event)
         _log.info("Event %i", len(self.events))
 
-        # This will extract info from events
+        # This will extract info from events.
         self.post_event()
 
-        # Check if we have enough events already
-        if len(self.events) >= self.n and self.n > 0:
+        # Check if we have enough events already.
+        if len(self.events) >= self.n > 0:
             self.fig.canvas.stop_event_loop()
 
     def post_event(self):
-        """For baseclass, do nothing but collect events"""
-        pass
```

Comment:
Is dropping the "pass" a python 3 thing?

Reply:
no, the docstring has always been enough.